src/Model/Linsen.hs

```diff
@@ -72,6 +69,20 @@ parse day =
           >=> (.: "richText")
           >=> (.: "root")
           >=> (.: "children")
+          >=> (\v' ->
+                 (case v' !? 1 of
+                   Nothing -> fail "failed to index into food"
+                   Just v -> pure v) >>=
+                 withObject "Parse day" (
+                  (.: "children")
+                  >=> (\case
+                          Nothing -> fail "Failed to index into richtext"
+                          Just v -> pure v) . headMay
+                  >=> (.: "text")
+                  >=> pure . (== pure day) . parseTime defaultTimeLocale "%d-%m-%Y"
+                  >=> \case
+                          True -> pure v'
+                          False -> pure []))
           >=> menuParser . (\v' -> if length v' >= 9 then v' else mempty)
```

Comment:
Also, due to me being way too tired my brain fired up the pattern recognition module and went to town, so the changes proposed below are kinda optional. :D 
```suggestion
                  >=> \s -> if pure day == parseTime defaultTimeLocale "%d-%m-%Y" s
                               && length v' >= 9
                        then pure v'
                        else mempty))
          >=> menuParser
```

Reply:
Added this, but `pure mempty` instead of `mempty` since `menuParser` still expects a list (Test failed when it wasn't so I take their word for it)